test/reducers/course.spec.js

```diff
@@ -1,16 +1,225 @@
+      title: '',
+      description: '',
+      school: '',
+      term: '',
+      level: '',
+      subject: '',
+      expected_students: '0',
+      start: null,
+      end: null,
+      timeline_start: null,
+      timeline_end: null,
+      day_exceptions: '',
+      weekdays: '0000000',
+      editingSyllabus: false
+    };
+
+    const newState = course(undefined, { type: null });
+    expect(newState).to.deep.eq(expectedState);
+  });
+
+  it('should handle RECEIVE_COURSE', () => {
+    const initialState = {};
+    const mockedAction = {
+      type: RECEIVE_COURSE,
+      data: { course: { title: 'title' } }
+    };
+    deepFreeze(initialState);
+
+    const newState = course(initialState, mockedAction);
+    expect(newState).to.deep.eq(mockedAction.data.course);
+  });
+
+  it('should handle RECEIVE_COURSE_UPDATE with newStats', () => {
+    const initialState = {};
+    const mockedAction = {
+      type: RECEIVE_COURSE_UPDATE,
+      data: { course: { title: 'title' } }
+    };
+    deepFreeze(initialState);
+
+    const newState = course(initialState, mockedAction);
+    expect(newState.title).to.deep.eq(mockedAction.data.course.title);
+
+    Object.keys(newState.newStats).forEach((key) => {
+      expect(newState.newStats[key]).to.eq(false);
+    });
+  });
+
+  it('should handle PERSISTED_COURSE', () => {
+    const initialState = { description: 'initial description' };
+    const mockedAction = {
+      type: PERSISTED_COURSE,
+      data: { course: { title: 'new title' } }
+    };
+    deepFreeze(initialState);
+
+    const newState = course(initialState, mockedAction);
+    expect(newState).to.deep.eq({
+      description: 'initial description',
+      title: 'new title'
+    });
+  });
+
+  it('should handle UPDATE_COURSE', () => {
     const initialState = { title: 'old title', term: 'old term' };
-    const action = { type: UPDATE_COURSE, course: { title: 'new title' } };
+    const mockedAction = {
+      type: UPDATE_COURSE,
+      course: { title: 'new title' }
+    };
+    deepFreeze(initialState);
+
+    const newState = course(initialState, mockedAction);
+    expect(newState).to.deep.eq({ title: 'new title', term: 'old term' });
+  });
+
+  it('should handle CREATED_COURSE', () => {
+    const initialState = {};
+    const mockedAction = {
+      type: CREATED_COURSE,
+      data: { course: { title: 'title' } }
+    };
+    deepFreeze(initialState);
+
+    const newState = course(initialState, mockedAction);
+    expect(newState.title).to.eq('title');
+  });
+
+  it('should handle RECEIVE_INITIAL_CAMPAIGN', () => {
+    const initialState = {};
+    const campaign = {
+      id: 2,
+      title: 'title',
+      template_description: 'description',
+      default_course_type: 'foobar',
+      default_passcode: 'pass'
+    };
+    const mockedAction = {
+      type: RECEIVE_INITIAL_CAMPAIGN,
+      data: { campaign: campaign }
+    };
+    deepFreeze(initialState);
+
+    const expectedState = {
```

Comment:
The idea here is that it uses the initial campaign to provide a default course description, type, and passcode, and also adds the properties for the initial campaign id and title.

Instead of expecting deep equality using the same code as from the reducer — which ties it closely to the implementation — this should pass in an initial state that is more like `{ title: 'default title', description: 'default description', type: 'default type', passcode: 'default passcode' }, and then check that the description, type and passcode got changed, the title was't changed, and the new properties are there and have the expected values.

Avoiding `deep.eq` in this case is a way to keep the test decoupled from the precise implementation, and will make the code more amenable to change without breaking the test.